implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheusmetrics/DefaultExemplarSamplerFactory.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.prometheusmetrics;
+
+import io.prometheus.metrics.config.ExemplarsProperties;
+import io.prometheus.metrics.core.exemplars.ExemplarSampler;
+import io.prometheus.metrics.core.exemplars.ExemplarSamplerConfig;
+import io.prometheus.metrics.tracer.common.SpanContext;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Default implementation of {@link ExemplarSamplerFactory}.
+ *
+ * @author Jonatan Ivanov
+ * @since 1.13.0
+ */
+public class DefaultExemplarSamplerFactory implements ExemplarSamplerFactory {
+
+    // TODO: Should we expose ExemplarsProperties and let users inject it through a ctor?
+    private final ExemplarsProperties properties = ExemplarsProperties.builder().build();
```

Comment:
Since we aren't implementing the behavior controlled by this, I think it's okay for users to rely on the configuration methods documented by the Prometheus client: https://prometheus.github.io/client_java/config/config/
As long as our usage allows these methods to work, then I don't think we need to expose something specifically.

Reply:
We don't allow changing these defaults and no one asked for this for the old client either.

Reply:
I looked into the implementation a bit. I assumed that using the builder would load the values from the locations specified in the documentation and use those unless programmatically overwritten. That's not the case. To get that sort of behavior, we should be using `PrometheusPropertiesLoader`, I think.

Reply:
`PrometheusProperties` keeps a singleton instance created by `PrometheusPropertiesLoader` and it's `get` method returns this instance so using that should be as simple as:
```java
private static final ExemplarsProperties EXEMPLARS_PROPERTIES = PrometheusProperties.get().getExemplarProperties();
```
and
```java
private static final ExporterProperties EXPORTER_PROPERTIES = PrometheusProperties.get().getExporterProperties();
```

But to me the question is more like: should we really do this and depend on PrometheusProperties? Or instead we should expose these properties (if we want to) in `PrometheusConfig`? In the case of `ExemplarsProperties` this might make some sense since we are using the defaults and this let the users override them (though it would be a valid question why they cannot override it through `PrometheusConfig`).
But in the case of `ExporterProperties` we are not using the default (`exemplarsOnAllMetricTypes` is false by default) so either we use the defaults and drop the extra exemplars by default or figure out how to let the users override this by using `ExporterProperties`.

Reply:
I did this here: https://github.com/micrometer-metrics/micrometer/pull/4867/commits/228d8598a47caa19bbc11a150aa8253424f284fb
With this, you can override `ExemplarsProperties` but not `ExporterProperties`.